crates/storage-query-datafusion/src/context.rs

```diff
@@ -99,86 +98,169 @@ pub trait SelectPartitions: Send + Sync + Debug + 'static {
     async fn get_live_partitions(&self) -> Result<Vec<(PartitionId, Partition)>, GenericError>;
 }
 
-#[derive(Clone)]
-pub struct QueryContext {
-    sql_options: SQLOptions,
-    datafusion_context: SessionContext,
+#[async_trait]
+pub trait TableRegisterer: Send + Sync + 'static {
+    async fn register(&self, ctx: &QueryContext) -> Result<(), BuildError>;
+}
```

Comment:
Maybe "RegisterTable"?  Alternatively, `TableRegistry` seems appropriate in this scenario too.